torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -3121,6 +3121,38 @@ def test_wait_all_with_partial_exception(self):
         with self.assertRaisesRegex(ValueError, "Expected error"):
             ret = torch.futures.wait_all(futs)
 
+    @dist_init(setup_rpc=False)
+    def test_init_rpc_twice(self):
+        initialize_pg(self.init_method, self.rank, self.world_size)
```

Comment:
It looks like we missed testing this, though it looks like this test fails depending on the init_method. For example if users want to init RPC twice with TCPStore, it currently does not work (see details in https://github.com/pytorch/pytorch/issues/46491).